### PR TITLE
Implement functions/commands for filtering find-file and insert

### PR DESCRIPTION
**Pull request by Zaeph**
Rejected for `org-roam` (jethrokuan/org-roam#481), but might be convenient for BibTeX users.

Upstream merge for jethrokuan/org-roam#481 is required.

### CODE REVIEW

**Comment by Zaeph** [comment]
jethrokuan/org-roam#481 has been merged, so we're good to go for the merge.

Any thoughts on this, @myshevchuk?

File: org-roam-bibtex.el [hunk 1]
```diff
@@ -447,5 +463,21 @@ notes project before calling any org-roam functions."
             (org-roam-find-file title))
         (message "Something went wrong. Check the *Warnings* buffer.")))))
 
+;;;###autoload
+(defun org-roam-find-non-ref-file (&optional initial-prompt)
+  "Find and open an Org-roam, non-ref file.
```
Comment (myshevchuk) [inline org-roam-bibtex.el:468]: What is "non-ref"?

File: org-roam-bibtex.el [hunk 2]
```diff
@@ -447,5 +463,21 @@ notes project before calling any org-roam functions."
...
+See `org-roam-find-files' and
+`org-roam--get-non-ref-path-completions' for details."
+  (interactive)
+  (org-roam-find initial prompt #'org-roam--get-non-ref-path-completions))
```
[the remark this answers is not in the document]
Comment (Zaeph) [reply org-roam-bibtex.el:472]: Whoops, thanks for catching that.

File: org-roam-bibtex.el (hunk 1 above)
Comment (Zaeph) [reply org-roam-bibtex.el:468]: Files which are not reference-notes, i.e. the ones without a `#+ROAM_KEY:`.

**Comment by Zaeph** [comment]
> Am I right that calling these functions will filter off the bibliography notes?

Yes.  The rationale is a bit clearer in jethrokuan/org-roam#481.

File: org-roam-bibtex.el (hunk 1 above)
Comment (myshevchuk) [reply org-roam-bibtex.el:468]: will be kept for display, right?

Comment (Zaeph) [reply org-roam-bibtex.el:468]: It's just an extra command to allow people to only show the notes which are not references.  One person in our tracker worried that his references would clutter up his `org-roam-find-file` selection.

**Comment by myshevchuk** [comment]
Yeah, this is a good start. I can envision something like more general `org-roam-bibtex-find-file` `org-roam-bibtex-insert` that when called with different universal arguments would apply different filter functions. 

I think it can be merged for now.

**Comment by Zaeph** [comment]
> Yeah, this is a good start. I can envision something like more general `org-roam-bibtex-find-file` `org-roam-bibtex-insert` that when called with different universal arguments would apply different filter functions.

Oh yeah, we're literally one step away from making shiftable or a `dwim` commands for those.

> I think it can be merged for now.

Great, thanks for reviewing it!

**Pull request merged**